spec/markdown-preview-view-spec.coffee

```diff
@@ -210,6 +210,31 @@ describe "MarkdownPreviewView", ->
         expect(fs.isFileSync(outputPath)).toBe true
         expect(atom.workspace.getActiveTextEditor().getText()).toBe expectedOutput
 
+    describe "text editor style extraction", ->
+
+      disposableStyleSheet = disposableStyleSheet2 = extractedStyles = null
+      textEditorStyle = ".editor-style .extraction-test { color: blue; }"
+      unrelatedStyle  = ".something else { color: red; }"
+
+      beforeEach ->
+        disposableStyleSheet = atom.styles.addStyleSheet textEditorStyle,
+          context: 'atom-text-editor'
+
+        disposableStyleSheet2 = atom.styles.addStyleSheet unrelatedStyle,
+          context: 'unrelated-context'
+
+        extractedStyles = preview.getTextEditorStyles()
+
+      it "returns an array containing atom-text-editor css style strings", ->
+        expect(extractedStyles.indexOf(textEditorStyle)).toBeGreaterThan(-1)
+
+      it "does not return other styles", ->
+        expect(extractedStyles.indexOf(unrelatedStyle)).toBe(-1)
+
+      afterEach ->
```

Comment:
Did you try not disposing of these. I'm not sure it should be needed because we restore the styles after each spec in the global spec helper.